core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -89,7 +89,44 @@ private[spark] class AppStatusStore(
     } else {
       base
     }
-    filtered.asScala.map(_.info).filter(_.id != FALLBACK_BLOCK_MANAGER_ID.executorId).toSeq
+    filtered.asScala.map(_.info)
+      .filter(_.id != FALLBACK_BLOCK_MANAGER_ID.executorId)
+      .map(replaceExec).toSeq
+  }
+
+  def replaceExec(origin: v1.ExecutorSummary): v1.ExecutorSummary = {
```

Comment:
Do we need to add a new public API? and we should add some comments

Reply:
make it private